Customize Jetpack connection signup screen for woocommerce core profiler flow

chihsuan opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to #77888 

## Proposed Changes

Customize Jetpack connection **signup** screen for woocommerce core profiler flow

![Screenshot 2023-06-09 at 15 28 15](https://github.com/Automattic/wp-calypso/assets/4344253/f9400241-78f1-4e86-a1a5-97b1fd842ed5)


## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Set up [woocommerce-start-dev-env](https://github.com/woocommerce/woocommerce-start-dev-env)
* Create a JN site
* Go to OBW and continue until Jetpack connection step (/jetpack/connect/authorize)
* Log out your wp account or open the current tab in an incognito window
* Change query parameter `from` to `woocommerce-core-profiler` and add `installed_ext_success=1` query parameter
* Observe the `Extensions successfully installed!` notice is shown and automatically dismissed
* Confirm the screen matches the design (Y5pUYSJPsGEud1vknUZhi8-fi-739_54680)
* Reload the page
* Observe the `Extensions successfully installed!` is not shown.
* Enable track debugging `localStorage.setItem( 'debug', 'calypso:analytics' );`

**Scenario: User clicks on No, Thanks**

1. Clicks on `No, Thanks` button
2. You should be redirected to your JN site
3.  Observe that `calypso_wc_coreprofiler_jpc_skip` event is recorded

**Scenario: User creation fails with validation errors**

1. Go back to JPC Signup page
2. Enter an invalid email address or username
3. Observe that the error message is shown under each form field

**Scenario: User clicks on Create an account button**

1. Clicks on `Create an account` button
2. Create a new account
3. Observe that `calypso_jpc_create_account` event is recorded
4. Should be redirected to the Jetpack connection screen
5. Should be authorized automatically (see the button is loading)
7. You should be redirected to your site
8. Log in if you're not logged in to your JN site
9. You should be redirected to `WooCommerce > Home`
10. Go to `Jetpack > My Jetpack` and Confirm the Jetpack is connected

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

chihsuan committed:
Customize jetpack connection signup screen for woocommerce core profiler flow

chihsuan committed:
Update button text and add jetpack logo to woo jpc signup page

chihsuan committed:
Update woo core profiler form border color

ilyasfoo commented on client/jetpack-connect/signup.js:
```diff
@@ -119,11 +120,16 @@ export class JetpackSignup extends Component {
 	};
 
-	isWoo() {
+	isWooOnboarding() {
```
As a person without much knowledge of Woo will find it confusing between this and `isWooCoreProfiler`. I think we can rename it to a name that signifies that it's an older version (or deprecated). Maybe, `isWooDeprecatedOnboarding` or `isWooLegacyOnboarding`?

chihsuan replied:
I agree, but I was thinking of changing this after we release the core profiler because it has not been deprecated yet. And we also need to deprecate all `isJetpackWooCommerceFlow `. What do you think?

ilyasfoo replied:
That makes sense, let's deprecate everything on a follow-up! 👍

ilyasfoo commented on client/jetpack-connect/woo-install-ext-success-notice.tsx:
```diff
@@ -0,0 +1,27 @@
+			showDismiss={ false }
+			text={ translate( 'Extensions successfully installed!' ) }
+			isCompact={ false }
+			duration={ 3000 }
```
This feels too short. What do you think?
```suggestion
			duration={ 6000 }
```

chihsuan replied:
Sounds good!

ilyasfoo reviewed (changes requested):
This is looking great, awesome work @chihsuan!

> Observe the Extensions successfully installed! notice is shown and automatically dismissed

The snackbar seems to appear below the visible screen when the screen height is considerably smaller. I think we may need to add fixed positioning to the snackbar. Unfortunately, it's not something the [component](https://wpcalypso.wordpress.com/devdocs/client/components/notice/README.md) already supports. It might be better to address this in a follow-up.

<img width="500" alt="image" src="https://github.com/Automattic/wp-calypso/assets/3747241/641613c4-3403-48b2-b83c-c3c696d3a82d">

> **Scenario: User clicks on Create an account button**
> ...
> 5. You should be redirected to your site
> 6. Log in if you're logged in in your JN site
> 

I'm a bit confused by this scenario. Are we using the JN site created previously? When creating a new account, the new account doesn't have admin access to JN, how does that work? 🤔 

In the end I got sent to the JN's wp-admin login screen, and not sure how to login with the new account since the only access available is via the JN admin account.

Some minor comments on the UI:

1. Let's set the max-width for the container to 615px according to figma (screenshot)
2. The terms text has slightly taller margin than design ([screenshot](https://github.com/Automattic/wp-calypso/assets/3747241/f85b66e4-c5ea-4f92-ae6c-add7b37817b9))
3. The username label in design is `CHOOSE A USERNAME`

chihsuan committed:
Update client/jetpack-connect/woo-install-ext-success-notice.tsx

Co-authored-by: Ilyas Foo <foo.ilyas@gmail.com>

chihsuan commented:
> I'm a bit confused by this scenario. Are we using the JN site created previously? When creating a new account, the new account doesn't have admin access to JN, how does that work? 
> In the end I got sent to the JN's wp-admin login screen, and not sure how to login with the new account since the only access available is via the JN admin account.

Thanks for the review! @ilyasfoo You can log in using your JN `demo` user account. 🙂  I'll address the feedback soon.

chihsuan committed:
Update jetpack connect woo core profiler notice position

chihsuan committed:
Address jpc woo UI feedback

chihsuan commented:
> The snackbar seems to appear below the visible screen when the screen height is considerably smaller. I think we may need to add fixed positioning to the snackbar. Unfortunately, it's not something the [component](https://wpcalypso.wordpress.com/devdocs/client/components/notice/README.md) already supports. It might be better to address this in a follow-up.

Good catch! I changed the CSS position to `fixed` in 5b3c30003fcdba56ca0268d2e6fb7b368be252a4.

> 1. Let's set the max-width for the container to 615px according to figma (screenshot)
> 2. The terms text has slightly taller margin than design ([screenshot](https://github.com/Automattic/wp-calypso/assets/3747241/f85b66e4-c5ea-4f92-ae6c-add7b37817b9))
> 3. The username label in design is `CHOOSE A USERNAME`

Thanks, updated the style and fixed the label in 063eba29e9906d13cbf4cc3ea049858eeb3e0d2a

ilyasfoo reviewed (approved):
> Good catch! I changed the CSS position to fixed in https://github.com/Automattic/wp-calypso/commit/5b3c30003fcdba56ca0268d2e6fb7b368be252a4.

Thanks for addressing, @chihsuan! One nitpick, I find the vertical distance between the snackbar and window is too far. Let's use the measurements from Figma ([screenshot](https://github.com/Automattic/wp-calypso/assets/3747241/325fcd5a-107b-43ab-b2c3-54d41067c621))

<img width="446" alt="image" src="https://github.com/Automattic/wp-calypso/assets/3747241/0e1f21a7-928b-44a2-a0f9-2b7280900c2a">

> Thanks for the review! @ilyasfoo You can log in using your JN demo user account. 🙂 I'll address the feedback soon.

Ok, understood! I tested using incognito and checked the Jetpack connection with my main browser instead.

Tested well, pre-approving!

moon0326 reviewed (approved):
Sorry for the delay in review 🙏 

LGTM and tested well 👍 🚀

chihsuan committed:
Fix notice vertical distance

chihsuan commented:
> > Good catch! I changed the CSS position to fixed in [5b3c300](https://github.com/Automattic/wp-calypso/commit/5b3c30003fcdba56ca0268d2e6fb7b368be252a4).
> 
> Thanks for addressing, @chihsuan! One nitpick, I find the vertical distance between the snackbar and window is too far. Let's use the measurements from Figma ([screenshot](https://github.com/Automattic/wp-calypso/assets/3747241/325fcd5a-107b-43ab-b2c3-54d41067c621))
> 
> <img alt="image" width="446" src="https://user-images.githubusercontent.com/3747241/245130789-0e1f21a7-928b-44a2-a0f9-2b7280900c2a.png">
> > Thanks for the review! @ilyasfoo You can log in using your JN demo user account. 🙂 I'll address the feedback soon.
> 
> Ok, understood! I tested using incognito and checked the Jetpack connection with my main browser instead.
> 
> Tested well, pre-approving!

Fixed vertical distance in 55b27f1c051b0918b9010003e5128f9c5764c1a8

ilyasfoo reviewed (approved):
LGTM! 🚢
